--- fortran/src/CMakeLists_orig.txt	2026-07-05 19:55:59
+++ fortran/src/CMakeLists.txt	2026-07-05 19:56:06
@@ -399,6 +399,9 @@
       LINKER_LANGUAGE Fortran
       Fortran_MODULE_DIRECTORY ${CMAKE_Fortran_MODULE_DIRECTORY}/static
   )
+  if (CMAKE_Fortran_COMPILER_ID STREQUAL "LLVMFlang")
+    set_target_properties (${HDF5_F90_LIB_TARGET} PROPERTIES LINKER_LANGUAGE C)
+  endif ()
   H5_SET_LIB_OPTIONS (${HDF5_F90_LIB_TARGET} ${HDF5_F90_LIB_NAME} STATIC 0)
   set_global_variable (HDF5_LIBRARIES_TO_EXPORT "${HDF5_LIBRARIES_TO_EXPORT};${HDF5_F90_LIB_TARGET}")
   set (install_targets ${install_targets} ${HDF5_F90_LIB_TARGET})
@@ -435,6 +438,9 @@
       LINKER_LANGUAGE Fortran
       Fortran_MODULE_DIRECTORY ${CMAKE_Fortran_MODULE_DIRECTORY}/shared
   )
+  if (CMAKE_Fortran_COMPILER_ID STREQUAL "LLVMFlang")
+    set_target_properties (${HDF5_F90_LIBSH_TARGET} PROPERTIES LINKER_LANGUAGE C)
+  endif ()
   set_global_variable (HDF5_LIBRARIES_TO_EXPORT "${HDF5_LIBRARIES_TO_EXPORT};${HDF5_F90_LIBSH_TARGET}")
   H5_SET_LIB_OPTIONS (${HDF5_F90_LIBSH_TARGET} ${HDF5_F90_LIB_NAME} SHARED "F")
   set (install_targets ${install_targets} ${HDF5_F90_LIBSH_TARGET})
--- hl/fortran/src/CMakeLists_orig.txt	2026-07-05 19:59:40
+++ hl/fortran/src/CMakeLists.txt	2026-07-05 19:59:04
@@ -187,6 +187,9 @@
       LINKER_LANGUAGE Fortran
       Fortran_MODULE_DIRECTORY ${CMAKE_Fortran_MODULE_DIRECTORY}/static
   )
+  if (CMAKE_Fortran_COMPILER_ID STREQUAL "LLVMFlang")
+    set_target_properties (${HDF5_HL_F90_LIB_TARGET} PROPERTIES LINKER_LANGUAGE C)
+  endif ()
   H5_SET_LIB_OPTIONS (${HDF5_HL_F90_LIB_TARGET} ${HDF5_HL_F90_LIB_NAME} STATIC 0)
   set_global_variable (HDF5_LIBRARIES_TO_EXPORT "${HDF5_LIBRARIES_TO_EXPORT};${HDF5_HL_F90_LIB_TARGET}")
   set (install_targets ${install_targets} ${HDF5_HL_F90_LIB_TARGET})
@@ -222,6 +225,9 @@
       LINKER_LANGUAGE Fortran
       Fortran_MODULE_DIRECTORY ${CMAKE_Fortran_MODULE_DIRECTORY}/shared
   )
+  if (CMAKE_Fortran_COMPILER_ID STREQUAL "LLVMFlang")
+    set_target_properties (${HDF5_HL_F90_LIBSH_TARGET} PROPERTIES LINKER_LANGUAGE C)
+  endif ()
   H5_SET_LIB_OPTIONS (${HDF5_HL_F90_LIBSH_TARGET} ${HDF5_HL_F90_LIB_NAME} SHARED "HL_F")
   set_global_variable (HDF5_LIBRARIES_TO_EXPORT "${HDF5_LIBRARIES_TO_EXPORT};${HDF5_HL_F90_LIBSH_TARGET}")
   set (install_targets ${install_targets} ${HDF5_HL_F90_LIBSH_TARGET})